src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -462,16 +441,18 @@ void SpatialPooler::initialize(
 }
 
 
-void SpatialPooler::compute(const SDR &input, const bool learn, SDR &active) {
+vector<SynapseIdx> SpatialPooler::compute(const SDR &input, const bool learn, SDR &active) const {
```

Comment:
WIP compute() is const. 
But that will not be feasible. (adaptSegment -> conn.updateSynapsePermanence)